### PR TITLE
Add: Support server-side grading.

### PR DESCRIPTION
**Pull request by bjones1**
This is the prereq before https://github.com/RunestoneInteractive/RunestoneServer/pull/1128 will pass.

### CODE REVIEW

File: runestone/common/project_template/conf.tmpl [hunk 1]
```diff
@@ -107,6 +107,16 @@ rst_prolog = (
...
+# Select whether to use server-side grading where possible. Server-side grading
+# requires **all** the following:
+#
+# - The use of Runestone services (``eBookConfig.useRunestoneServices === true``)
+# - A valid login (``eBookConfig.loginRequired === true``)
+# - Logging enabled (``eBookConfig.logLevel  > 0``)
+#
+# The first two conditions cause the ``RunestoneBase.logBookEvent`` in ``runestonebase.js`` to post a student response to the server. The last conditions ensures that ``hsblog`` in ``ajax.py`` on the server will return a response containing grading information.
+runestone_server_side_grading = %(server_side_grading)s
+
```
Comment (bnmnetp) [inline runestone/common/project_template/conf.tmpl]: OK, so this is the explicit parameter to enable server side grading?

**Comment by bjones1** [comment]
OK -- I'll so some manual testing, and perhaps add a unit test for the timed exams.

File: runestone/common/project_template/conf.tmpl (hunk 1 above)
Comment (bjones1) [reply runestone/common/project_template/conf.tmpl]: Yes, for the Sphinx generation code/code in this repo. The server-side code (ajax.py) also needs to know that server-side grading is enabled, but (obviously) that needs to be discussed/addressed at https://github.com/RunestoneInteractive/RunestoneServer/pull/1128.

**Comment by bjones1** [comment]
I'd also like to implement this for multiple-choice questions, but thought it would be best just to get one question type up and running first. I've haven't implemented mchoice yet.

**Comment by bjones1** [comment]
OK, this one is ready. It includes timed assessment tests and fixes.

**Comment by bjones1** [comment]
I just rebased and pushed a fix to make this work in tabbed assessments. It's ready to go.

**Comment by bnmnetp** [comment]
@bjones1 I will try to get to this today or tomorrow.  In a bit of an accessibility fire drill right now.

Does you school have rules about WCAG compliance for web based tools in the classroom?

**Comment by bjones1** [comment]
I understand. Re: WCAG: no, not to my knowledge -- I had to actually look up the term. I can't find much on it on the MSU web site, either.

**Comment by bnmnetp** [comment]
More and more large public universities are adopting policies that web based tools used in the classroom must be WCAG 2.0 compliant.  I get it, and I support it, but its confusing and potentially a lot of work.

**Comment by bjones1** [comment]
I understand. Sigh...that sounds like a BUNCH of work...

If I can help, let me know. It's probably worth doing.

**Comment by bjones1** [comment]
Pinging...

**Comment by bjones1** [comment]
Pinging...

**Comment by bjones1** [comment]
This won't work until a field is added to the database. Since this change belongs in the server, this PR isn't ready yet.

**Comment by bjones1** [comment]
Closing until the server side is updated.

**Pull request closed without merging**